ir/instr.cpp

```diff
@@ -567,8 +567,8 @@ expr BinOp::getTypeConstraints(const Function &f) const {
                   lhs->getType() == rhs->getType();
 
     if (auto ty = getType().getAsStructType()) {
-      instrconstr &= ty->numElements() == 2 &&
-                     ty->getChild(0) == lhs->getType() &&
+      // Note that ty->numElements() may not be 2 due to paddings
```

Comment:
this is dangerous. can't we have a solution like the number of elements is two modulo padding?

Reply:
We have the number of padding as constant only so I didn't use the condition.
Would it be helpful to have the guard still?

Reply:
The check revived